crates/chain/src/keychain/txout_index.rs

```diff
@@ -61,10 +135,14 @@ const DEFAULT_LOOKAHEAD: u32 = 1_000;
 #[derive(Clone, Debug)]
 pub struct KeychainTxOutIndex<K> {
     inner: SpkTxOutIndex<(K, u32)>,
```

Comment:
```suggestion
    inner: SpkTxOutIndex<(DescriptorId, u32)>,
```

Because it is possible to change the descriptor of a keychain, this is a more accurate representation.

Reply:
Because of this, we may need two versions of some methods.

I.e. `reveal_to_target` will have a non-pub version: `reveal_to_target_with_id` that takes in a `DescriptorId` instead of `K`.

Reply:
Updated, I don't think we need a non-pub version of methods that takes descriptor_id as the code is kind of clean now already, what do you think?

Reply:
Mhh. Actually, reveal_to_target is called in a couple of places where I had the descriptor_id available, and instead I'm calling it with the keychain and expecting on the result, so `reveal_to_target_with_id` would be clean. At the same time, reveal_to_target needs both the desc_id and the descriptor, and at the moment we don't have a quick way of returning a descriptor given the id.
What do you think it's better:
1. Leave the code as is
2. Create a desc_id -> desc map (we could also have a public method returning this info?)
3. Change the descriptor_ids_to_keychains to be (desc_id -> (desc, keychain)) instead of (desc_id -> keychain). Makes the code way less readable imho, as we need to call `.1` almost everywhere to only retrieve the keychain 

I'm thinking 2 but what do you think?

Reply:
I'm actually in favor of 3. Instead of `.1`, we can have `let (_, keychain) = self.descriptor_ids_to_keychain.get(something)`. I think having less fields makes it easier to maintain and reason about.

However, I would be fine with 1 or 2 (although I would prefer 1 over 2).